modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/suspended/ConcurrentSyntax.kt

```diff
@@ -0,0 +1,103 @@
+package arrow.effects.typeclasses.suspended
+
+import arrow.Kind
+import arrow.core.Tuple2
+import arrow.core.Tuple3
+import arrow.core.Tuple4
+import arrow.core.Tuple5
+import arrow.effects.typeclasses.Concurrent
+import arrow.effects.typeclasses.Fiber
+import arrow.effects.typeclasses.RacePair
+import arrow.effects.typeclasses.RaceTriple
+import kotlin.coroutines.CoroutineContext
+
+interface ConcurrentSyntax<F> : AsyncSyntax<F>, Concurrent<F>, ListParTraverseSyntax<F> {
+
+  private suspend fun <A> concurrently(fb: Concurrent<F>.() -> Kind<F, A>): A =
+    run<Concurrent<F>, Kind<F, A>> { fb(this) }.bind()
+
+  suspend fun <A> asyncCallback(k: SProc<A>): A =
+    concurrently { asyncF(k.kr()) }
+
+  override suspend fun <A> CoroutineContext.startFiber(f: suspend () -> A): Fiber<F, A> =
+    concurrently { f.k().startF(this@startFiber) }
+
+  suspend fun <A, B> CoroutineContext.racePair(
+    fa: suspend () -> A,
+    fb: suspend () -> B
+  ): RacePair<F, A, B> =
+    concurrently { racePair(this@racePair, fa.k(), fb.k()) }
+
+  suspend fun <A, B, C> CoroutineContext.raceTriple(
+    fa: suspend () -> A,
+    fb: suspend () -> B,
+    fc: suspend () -> C
+  ): RaceTriple<F, A, B, C> =
+    concurrently { raceTriple(this@raceTriple, fa.k(), fb.k(), fc.k()) }
+
+  suspend fun <A, B, C> CoroutineContext.parMap(
+    fa: suspend () -> A,
+    fb: suspend () -> B,
+    f: (A, B) -> C
+  ): C =
+    concurrently { parMapN(this@parMap, fa.k(), fb.k(), f) }
+
+  suspend fun <A, B, C, D> CoroutineContext.parMap(
+    fa: suspend () -> A,
+    fb: suspend () -> B,
+    fc: suspend () -> C,
+    f: (A, B, C) -> D
+  ): D =
+    concurrently { parMapN(this@parMap, fa.k(), fb.k(), fc.k(), f) }
+
+  suspend fun <A, B, C, D, E> CoroutineContext.parMap(
+    fa: suspend () -> A,
+    fb: suspend () -> B,
+    fc: suspend () -> C,
+    fd: suspend () -> D,
+    f: (A, B, C, D) -> E
+  ): E =
+    concurrently { parMapN(this@parMap, fa.k(), fb.k(), fc.k(), fd.k(), f) }
+
+  suspend fun <A, B, C, D, E, G> CoroutineContext.parMap(
+    fa: suspend () -> A,
+    fb: suspend () -> B,
+    fc: suspend () -> C,
+    fd: suspend () -> D,
+    fe: suspend () -> E,
+    f: (A, B, C, D, E) -> G
+  ): G =
+    concurrently { parMapN(this@parMap, fa.k(), fb.k(), fc.k(), fd.k(), fe.k(), f) }
+
+
+  suspend fun <A, B> CoroutineContext.parallel(
```

Comment:
The spirit of the Fx DSL is to offer even friendlier names. The use of this function does not even expose tuple to the end user because destructuring is automatic in Tuple. I think a new user may not even understand tuples all around but can understand the functions passed into `parallel` run in parallel and return their result independently for as many functions you use as args:
```kotlin
val (ta, tb) = parallel({ Thread.currentThread.name }, { Thread.currentThread.name })
```

Reply:
Once we move all the syntax to type classes then this will moved with it.